### PR TITLE
drop 3.6

### PR DESCRIPTION
**Pull request by xiangyan99**
# Description

Please add an informative description that covers that changes made by the pull request and link all relevant issues.

If an SDK is being regenerated based on a new swagger spec, a link to the pull request containing these swagger spec changes has been included above.

# All SDK Contribution checklist:
- [ ] **The pull request does not introduce [breaking changes]**
- [ ] **CHANGELOG is updated for new features, bug fixes or other significant changes.**
- [ ] **I have read the [contribution guidelines](https://github.com/Azure/azure-sdk-for-python/blob/main/CONTRIBUTING.md).**

## General Guidelines and Best Practices
- [ ] Title of the pull request is clear and informative.
- [ ] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For more information on cleaning up the commits in your PR, [see this page](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/cleaning-up-commits.md).

### [Testing Guidelines](https://github.com/Azure/azure-sdk-for-python/blob/main/CONTRIBUTING.md##building-and-testing)
- [ ] Pull request includes test coverage for the included changes.

### CODE REVIEW

File: eng/pipelines/templates/stages/platform-matrix-cryptography-dependency.json [hunk 1]
```diff
@@ -28,13 +28,6 @@
       },
       "PythonVersion": "3.7",
       "CoverageArg": "--disablecov"
-    },
```
Comment (scbedd) [inline eng/pipelines/templates/stages/platform-matrix-cryptography-dependency.json:31]: This entirely eliminates windows. You can remove one of the versions from above, and update the current target to whatever you choose.

**Review by scbedd — APPROVED** [review]
If you don't do any updates to `identity`, please file an issue so we can track it and get it resolved.

I've pushed updates to cover any matrixes you were unaware of.

File: eng/pipelines/templates/stages/platform-matrix.json [hunk 2]
```diff
@@ -10,7 +10,7 @@
       "windows-2019": { "OSVmImage": "MMS2019", "Pool": "azsdk-pool-mms-win-2019-general" },
       "macOS-10.15": { "OSVmImage": "macOS-10.15", "Pool": "Azure Pipelines" }
     },
-    "PythonVersion": [ "pypy3.7", "3.6", "3.7", "3.8" ],
+    "PythonVersion": [ "pypy3.7", "3.7", "3.8" ],
```
Comment (benbp) [inline eng/pipelines/templates/stages/platform-matrix.json:13]: Do you want to swap 3.6 for 3.9 here as well?

**Review by benbp — APPROVED** [review]

**Comment by scbedd** [comment]
The build failure is covered in #22498 

I will investigate and resolve outside of this PR. I recommend check enforcer override to unblock this merge. If I can't get a resolution today, I'll revert just the changes to the storage matrix until I can resolve the problem.

**Pull request merged**